tests/test_opencode_rust_coverage_toolchain_contract.py

```
@@ -0,0 +1,202 @@
+    """Require the trusted toolchain guard to fail closed before cargo llvm-cov."""
+
+    dispatch = _dispatch_text()
+    helper = _helper_text()
+    assert f'[ "${{LLVM_COV:-}}" != "{_LLVM_COV_PATH}" ]' in dispatch
+    assert f'[ "${{LLVM_PROFDATA:-}}" != "{_LLVM_PROFDATA_PATH}" ]' in dispatch
+    assert 'test -x "$LLVM_COV"' in dispatch
+    assert 'test -x "$LLVM_PROFDATA"' in dispatch
+    assert "networkless coverage runtime did not preserve the reviewed LLVM 19" in dispatch
+    assert f'LLVM_COV_PATH="{_LLVM_COV_PATH}"' in helper
+    assert f'LLVM_PROFDATA_PATH="{_LLVM_PROFDATA_PATH}"' in helper
+    assert f'"${{LLVM_COV:-}}" != "$LLVM_COV_PATH"' in helper
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)